compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/CallCompleter.kt

```diff
@@ -398,4 +401,13 @@ class CallCompleter(
         val expressionType = trace.getType(expression.receiverExpression)
         return expressionType != null && TypeUtils.isNullableType(expressionType)
     }
+
+    private fun MutableResolvedCall<*>.updateResultDataFlowInfoUsingEffects(bindingTrace: BindingTrace) {
+        val moduleDescriptor = DescriptorUtils.getContainingModule(this.resultingDescriptor?.containingDeclaration ?: return)
+
+        val resultDFIfromES = effectSystem.getResultDataFlowInfo(this, bindingTrace, moduleDescriptor)
+        this.dataFlowInfoForArguments.updateResultInfo(resultDFIfromES)
+
+        effectSystem.recordDefiniteInvocations(this, bindingTrace, moduleDescriptor)
```

Comment:
Introduction of separate language features required some refactoring of EffectSystem endpoints to make them more granular (previously, lambdas invocations were written during inference of resulting data flow info), thus requiring users of Effect System now ask for recording of definite invocations explicitly. Probably, commit message needs a bit of rewording.